src/raytracer/Raytracer.m

```diff
@@ -64,9 +64,20 @@
 % Modified by: Mattia Lecci <leccimat@dei.unipd.it>, Refactored code
 
 
-%% Input Parameters Management
-nodeLoc = nodeCfgInput.nodeLoc;
+%% Input Parameters Management and preallocation
+nodeLoc(1,:,:) = nodeCfgInput.nodeLoc;
+nodePosition = nodeCfgInput.nodePosition;
 nodeVelocities = nodeCfgInput.nodeVelocities;
+nPAA_centroids = cellfun(@(x) x.nPAA_centroids ,nodeCfgInput.PAA_info );
+Mpc = cell(paraCfgInput.numberOfNodes,...
+    max(nPAA_centroids),...
+    paraCfgInput.numberOfNodes,...
+    max(nPAA_centroids),...
+    paraCfgInput.totalNumberOfReflections+1,...
+    paraCfgInput.numberOfTimeDivisions+1 );
+frmRotMpInfo = cell(1, paraCfgInput.totalNumberOfReflections+1);
+keepBothQDOutput =0; % If 1 when using JSON output, it will keep also previous QD output
```

Comment:
maybe i would change paraCfgInput.jsonOutput to paraCfgInput.outputFormat = {'json', 'legacy', 'both'}, what do you think?